descriptors/src/legacy.rs

```diff
@@ -110,58 +109,128 @@ impl DerivePublicKey for SingleSig {
 impl MiniscriptKey for SingleSig {
     type Hash = Self;
 
-    fn to_pubkeyhash(&self) -> Self::Hash { self.clone() }
+    fn to_pubkeyhash(&self) -> Self::Hash {
+        self.clone()
+    }
 }
 
 impl FromStr for SingleSig {
     type Err = ComponentsParseError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        static ERR: &str =
-            "wrong build-in pubkey placeholder regex parsing syntax";
-
-        lazy_static! {
-            static ref RE_PUBKEY: Regex = Regex::new(
-                r"(?x)^
-                (\[
-                    (?P<fingerprint>[0-9A-Fa-f]{8})      # Fingerprint
-                    (?P<deviation>(/[0-9]{1,10}[h']?)+)  # Derivation path
-                \])?
-                (?P<pubkey>0[2-3][0-9A-Fa-f]{64}) |      # Compressed pubkey
-                (?P<pubkey_long>04[0-9A-Fa-f]{128})      # Non-compressed pubkey
-                $",
-            )
-            .expect(ERR);
-        }
-        if let Some(caps) = RE_PUBKEY.captures(s) {
-            let origin = if let Some((fp, deriv)) =
-                caps.name("fingerprint").map(|fp| {
-                    (fp.as_str(), caps.name("derivation").expect(ERR).as_str())
-                }) {
+        static ERR: &str = "wrong build-in pubkey parsing syntax";
```

Comment:
nit: I think we do not need this string as a static anymore and it can be just inlined below. We needed it before since it was used twice (but not anymore)